test/robot/job/worker/worker.go

```diff
@@ -107,6 +107,22 @@ func RetryFunction(ctx context.Context, maxAttempts int, delay time.Duration, ta
 	return log.Errf(ctx, err, "Retried maximum attempts, %v", maxAttempts)
 }
 
+func WithTimeout(ctx context.Context, timeout time.Duration, task func() error) error {
```

Comment:
This isn't how timeouts should be done - this will leave the task executing and the caller will return prematurely.
The idiomatic way is to use context cancellation - See `task.WithTimeout`

Reply:
Cool, just realized we already have all such infrastructure done. I was thinking how to kill the process in case it is cancelled.
DONE